src/modules/expression-evaluator/src/evaluateExpression.ts

```diff
@@ -1,149 +1,146 @@
-import { IConnection, IQueryNode, IParameters, IGraphQLConnection, BasicObject } from './types'
+import {
+  IConnection,
+  IQueryNode,
+  IParameters,
+  IGraphQLConnection,
+  NodeType,
+  BasicObject,
+} from './types'
 
 const defaultParameters: IParameters = {}
 
 export default async function evaluateExpression(
   inputQuery: IQueryNode | string | number | boolean | any[],
```

Comment:
or object ?

Reply:
Okay, in light of the changes here (and gradual evolution since this was born), I've cleaned up and simplified the types overall. 7a953d6d

Every node is now either a `OperatorNode` or a `ValueNode` type, defined as:

```
export interface OperatorNode {
  operator: Operator
  type?: OutputType
  children: Array<OperatorNode | ValueNode>
  value?: ValueNode // deprecated
}

export type ValueNode = string | boolean | number | BasicObject | null | undefined | any[]
```

The evaluator takes `OperatorNode | ValueNode` as input, and returns `Promise<ValueNode>`